src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubectlBuildWrapper.java

```diff
@@ -0,0 +1,129 @@
+import com.cloudbees.plugins.credentials.domains.URIRequirementBuilder;
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.AbstractProject;
+import hudson.model.Item;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.security.ACL;
+import hudson.tasks.BuildWrapperDescriptor;
+import hudson.util.ListBoxModel;
+import hudson.util.Secret;
+import jenkins.model.Jenkins;
+import jenkins.tasks.SimpleBuildWrapper;
+import org.kohsuke.stapler.AncestorInPath;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.Collections;
+
+/**
+ * @author <a href="mailto:nicolas.deloof@gmail.com">Nicolas De Loof</a>
+ */
+public class KubectlBuildWrapper extends SimpleBuildWrapper {
+
+    private final String serverUrl;
+    private final String credentialsId;
+
+    @DataBoundConstructor
+    public KubectlBuildWrapper(@Nonnull String serverUrl, String credentialsId) {
+        this.serverUrl = serverUrl;
+        this.credentialsId = credentialsId;
+    }
+
+
+    @Override
+    public void setUp(Context context, Run<?, ?> build, FilePath workspace, Launcher launcher, TaskListener listener, EnvVars initialEnvironment) throws IOException, InterruptedException {
+
+        FilePath configFile = workspace.createTempFile(".kube", "config");
+
+        launcher.launch()
+                .cmdAsSingleString("kubectl config --kubeconfig=" + configFile.getRemote() + " set-cluster k8s --server=" + serverUrl + " --insecure-skip-tls-verify=true")
+                .join();
+
+        final UsernamePasswordCredentials c = getCredentials(credentialsId);
+
+        launcher.launch()
+                .cmdAsSingleString("kubectl config --kubeconfig=" + configFile.getRemote() + " set-credentials cluster-admin --username=" + c.getUsername() + " --password=" + Secret.toString(c.getPassword()))
+                .masks(false, false, false, false, false, false, true)
+                .join();
+
+        launcher.launch()
+                .cmdAsSingleString("kubectl config --kubeconfig=" + configFile.getRemote() + " set-context k8s --cluster=k8s --user=cluster-admin")
+                .join();
+
+        launcher.launch()
+                .cmdAsSingleString("kubectl config --kubeconfig=" + configFile.getRemote() + " use-context k8s")
+                .join();
+
+        context.setDisposer(new CleanupDisposer(configFile));
+
+        context.env("KUBECONFIG", configFile.getRemote());
+    }
+
+    private UsernamePasswordCredentials getCredentials(String credentials) {
+        return CredentialsMatchers.firstOrNull(
+                CredentialsProvider.lookupCredentials(StandardUsernamePasswordCredentials.class,
+                        Jenkins.getInstance(), ACL.SYSTEM, Collections.<DomainRequirement>emptyList()),
+                CredentialsMatchers.withId(credentials)
```

Comment:
I assume `CredentialsMatchers.withId(credentials)` can handle null `credentials`?

An aside ... I've heard the theories/arguments on not needing to perform `null` checks where findbugs annotations are in use. I still don't buy any of them :smile: Nothing to do with this though.